src/main/java/mods/railcraft/common/util/inventory/AdjacentInventoryCache.java

```diff
@@ -81,4 +81,20 @@ public InventoryComposite getAdjacentInventories() {
         return sortedInvs;
     }
 
+    public InventoryComposite getInventoryOnSide(EnumFacing face) {
+        Map<EnumFacing, TileEntity> tiles = cache.refreshTiles();
+        if (changedSides.contains(face)) {
+            invs.remove(face);
+            TileEntity tile = tiles.get(face);
+            if (tile != null && (filter == null || filter.test(tile))) {
+                IInventoryObject inv = InventoryFactory.get(tile, face.getOpposite());
+                if (inv != null)
+                    invs.put(face, inv);
+            }
+
+            changedSides.clear();
+            sortedInvs.add(invs.get(face));
```

Comment:
You realize you are returning an ever growing list here? Every time the function is called, it adds a new entry. You removed the clear. Its also returning _all_ the other sides too.

Reply:
This one is actually unused. I will try to revert this change at best, since this cache can only pull inventory from all sides.